progress.go

```diff
@@ -11,27 +13,48 @@ const (
 )
 
 type TableProgress struct {
+	TableName                   string
+	PaginationKeyName           string
 	LastSuccessfulPaginationKey uint64
 	TargetPaginationKey         uint64
-	CurrentAction               string // Possible values are defined via the constants TableAction*
+	Status                      string // Possible values are defined via the constants TableAction*
 	RowsWritten                 uint64
 }
 
 type Progress struct {
+	GhostferryVersion string
+
+	SourceHostPort string
+	TargetHostPort string
+
+	StartTime   time.Time
+	CurrentTime time.Time
+	TimeTaken   time.Duration
+
+	AllTableNames    []string
+	AllDatabaseNames []string
+
+	CompletedTableCount int
+	TotalTableCount     int
+
 	// Possible values are defined in ferry.go
 	// Shows what the ferry is currently doing in one word.
 	CurrentState string
+	OverallState string
 
 	// The Payload field of the ProgressCallback config will be copied to here
 	// verbatim.
 	// Example usecase: you can be sending all the status to some aggregation
 	// server and you want some sort of custom identification with this field.
 	CustomPayload string
 
-	Tables                  map[string]TableProgress
-	LastSuccessfulBinlogPos mysql.Position
-	BinlogStreamerLag       float64 // seconds
-	Throttled               bool
+	Tables                      map[string]TableProgress
+	LastSuccessfulBinlogPos     mysql.Position
+	TargetBinlogPos             mysql.Position
+	BinlogStreamerLag           time.Duration // seconds
```

Comment:
Will this still show seconds over 60s? Doesn't duration format to something like `1m2s`?

Reply:
yeah you're correct. This comment should be removed as the BinlogStreamerLag is not in seconds.